### ISSUE TITLE
[✨] Multiple Flutter Versions in one System

### ISSUE THREAD

**Issue by ZiyadF296** [FlutterMatic/desktop#51]
**Is your feature request related to a problem? Please describe.**
Allow storing more than one Flutter channel on the system

**Describe the solution you'd like**
Store multiple Flutter channels at once.

**Describe alternatives you've considered**
Manually switching and basically reinstalling Flutter by checking out the current channel branch and checking in the new branch (takes time).

**Additional context**
- Storing multiple should be optional (default channel stable)
- This is planned to be available at a stable release directly (skipping alpha and beta)

**Comment by yahu1031** [comment]
This was already an existing application called [FVM(CLI)](https://fvm.app/). This will not be a good idea to overcomplicate the things WRT FVM. No one will use it cause everything with FVM is well aligned and assigned. So I'm not upvoting this feature.

**Comment by ZiyadF296** [comment]
Yes, we want to integrate this into FM to minimize the use of multiple tools. This makes it easier to switch between channels and so forth. Perhaps we could integrate this tool into FM.

**Comment by yahu1031** [comment]
Nope @ZiyadF296. FVM is not so hard to use. Actually, it is very lightweight and easy as other CLI apps. And it a not a useful thing. I can't say it is useless too but it is not useful. Please make sure to add a feature to a TODO task list when contributors are on the same ship. And if the feature is accepted then add an enhancement label to it. And add it to the TODO list of the next releases. Do not waste your time on doing that which is not at all required as of now.

**Comment by ZiyadF296** [comment]
I am not wasting my own time. Time is something I value myself. As mentioned earlier, FVM is not difficult or challenging to use. However, the idea of FM is to eliminate the use of CLI as much as possible when working with Flutter. Adding FVM to FM and allowing integration and usage via our UI would be a useful feature. If we are going to add FVM we will not write our channel switching code, instead, FVM handles all of this for us at its core. Instead, we will just integrate it to FM to use it underneath the hood.

It's a very simple feature and can be added easily, but since it's not that important and probably not many people will use it, it has been marked as `P4` and `not so important`. That way, whenever it is ready to be added (meaning the app is stable to accept new features), then this will be considered to be added.

**Comment by yahu1031** [comment]
I'm not upvoting it 😃 👎🏻

**Comment by ZiyadF296** [comment]
I created this issue as a way to collect opinions. Your opinion is appreciated :)
